extensions/cactus-plugin-htlc-coordinator/src/test/typescript/integration/plugin-htlc-coordinator/withdraw-counterparty-endpoint.test.ts

```diff
@@ -0,0 +1,295 @@
+    // signing credential pointing to this keychain entry.
+    backend: new Map([[TestTokenJSON.contractName, TestTokenJSON]]),
+    logLevel,
+  });
+  keychainPlugin.set(DemoHelperJSON.contractName, DemoHelperJSON);
+  keychainPlugin.set(HashTimeLockJSON.contractName, HashTimeLockJSON);
+
+  const factory = new PluginFactoryLedgerConnector({
+    pluginImportType: PluginImportType.Local,
+  });
+
+  const pluginRegistry = new PluginRegistry({});
+  const connector: PluginLedgerConnectorBesu = await factory.create({
+    rpcApiHttpHost,
+    rpcApiWsHost,
+    logLevel,
+    instanceId: connectorId,
+    pluginRegistry: new PluginRegistry({ plugins: [keychainPlugin] }),
+  });
+
+  const httpServer = createServer();
+  await new Promise((resolve, reject) => {
+    httpServer.once("error", reject);
+    httpServer.once("listening", resolve);
+    httpServer.listen(0, "127.0.0.1");
+  });
+
+  pluginRegistry.add(connector);
+  const pluginOptions: IPluginHTLCCoordinatorOptions = {
+    instanceId: uuidv4(),
+    logLevel,
+    pluginRegistry,
+  };
+
+  const factoryHTLC = new PluginFactoryHTLCCoordinator({
+    pluginImportType: PluginImportType.Local,
+  });
+  const pluginHTLCCoordinator = await factoryHTLC.create(pluginOptions);
+  pluginRegistry.add(pluginHTLCCoordinator);
+
+  const expressApp = express();
+  expressApp.use(bodyParser.json({ limit: "250mb" }));
+  const server = http.createServer(expressApp);
+  const listenOptions: IListenOptions = {
+    hostname: "0.0.0.0",
+    port: 0,
+    server,
+  };
+  const addressInfo = (await Servers.listen(listenOptions)) as AddressInfo;
+  test.onFinish(async () => await Servers.shutdown(server));
+  const { address, port } = addressInfo;
+  const apiHost = `http://${address}:${port}`;
+
+  const configuration = new Configuration({ basePath: apiHost });
+  const api = new BesuApi(configuration);
+
+  await pluginHTLCCoordinator.getOrCreateWebServices();
+  await pluginHTLCCoordinator.registerWebServices(expressApp);
+  t.comment("Deploys TestToken via .json file on deployContract function");
+  const deployOutToken = await connector.deployContract({
+    contractName: TestTokenJSON.contractName,
+    contractAbi: TestTokenJSON.abi,
+    bytecode: TestTokenJSON.bytecode,
+    web3SigningCredential,
+    keychainId,
+    constructorArgs: ["100", "token", "2", "TKN"],
+    gas: estimatedGas,
+  });
+  t.ok(deployOutToken, "deployContract() output is truthy OK");
+  t.ok(
+    deployOutToken.transactionReceipt,
+    "deployContract() output.transactionReceipt is truthy OK",
+  );
+  t.ok(
+    deployOutToken.transactionReceipt.contractAddress,
+    "deployContract() output.transactionReceipt.contractAddress is truthy OK",
+  );
+  const tokenAddress = deployOutToken.transactionReceipt
+    .contractAddress as string;
+
+  t.comment("Approve 10 Tokens to HashTimeLockAddress");
+  const { success } = await connector.invokeContract({
```

Comment:
@jagpreetsinghsasan Please assert more specifically the status code being 2xx and also refactor this to use the API client not the connector instance directly.